# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/AudioStream.h
# Commit: f604fe0f8d02
# Full Hash: f604fe0f8d0275c12a17fa492948871fdd9f59bf
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-10 03:50:31
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Wait for the audio callbacks to start being called before using the audio clock. r=alwu,media-playback-reviewers
#   
#   Because we're generally using high latency on the cubeb stream used by
#   AudioStream instance, it can take some time for the callbacks to start being
#   called, and the for the audio clock (cubeb_stream_get_position(...)) to advance.
# ==============================================================================

diff -r 61ec6368ddcd -r f604fe0f8d02 dom/media/AudioStream.h
--- a/dom/media/AudioStream.h	Mon May 09 13:20:50 2022 +0000
+++ b/dom/media/AudioStream.h	Mon May 09 13:20:50 2022 +0000
@@ -297,6 +297,9 @@
 
   bool IsPlaybackCompleted() const;
 
+  // Returns true if at least one DataCallback has been called.
+  bool CallbackStarted() const { return mCallbacksStarted; }
+
  protected:
   friend class AudioClock;
 
@@ -386,6 +389,7 @@
   std::atomic<bool> mPreservesPitch;
   // Audio thread only
   bool mAudioThreadChanged = false;
+  Atomic<bool> mCallbacksStarted;
 };
 
 }  // namespace mozilla